Dfe.PrepareConversions/Dfe.PrepareConversions/Pages/TaskList/GeneralInformation/PartOfPfiScheme.cshtml.cs

```diff
@@ -0,0 +1,60 @@
+using Dfe.PrepareConversions.Data;
+using Dfe.PrepareConversions.Data.Models;
+using Dfe.PrepareConversions.Data.Services;
+using Dfe.PrepareConversions.Extensions;
+using Dfe.PrepareConversions.Models;
+using Dfe.PrepareConversions.Pages;
+using Dfe.PrepareConversions.Services;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace Dfe.PrepareConversions.Pages.TaskList.GeneralInformation;
+
+public class PartOfPfiModel : BaseAcademyConversionProjectPageModel
+{
+   private readonly ErrorService _errorService;
+
+   public PartOfPfiModel(IAcademyConversionProjectRepository repository, ErrorService errorService) : base(repository)
+   {
+      _errorService = errorService;
+   }
+   public bool ShowError => _errorService.HasErrors();
+
+   [BindProperty]
+   public bool? YesChecked { get; set; }
+
+   [BindProperty]
+   public string PfiSchemeDetails { get; set; }
+
+   public override async Task<IActionResult> OnGetAsync(int id)
+   {
+      await base.OnGetAsync(id);
+
+      YesChecked = Project.PartOfPfiScheme.ToBool();
+      PfiSchemeDetails = Project.PfiSchemeDetails;
+
+      return Page();
+   }
+
+   public override async Task<IActionResult> OnPostAsync(int id)
+   {
+      if (YesChecked is true && string.IsNullOrWhiteSpace(PfiSchemeDetails))
+      {
+         ModelState.AddModelError(nameof(PfiSchemeDetails), "You must enter the PFI scheme the school is part of");
+      }
+
+      if (ModelState.IsValid)
+      {
```

Comment:
I don't know if this is possible on our project, but just mentioning it as a general thing.
I think it's better to have construction logic in a constructor, so that the responsibility around what is valid is owned by the abstraction.
(talking about PfiSchemeDetails = YesChecked is true ? PfiSchemeDetails : default)

This might be at odds with all the code already there, so just mentioning it.

Reply:
Yeah that makes sense! It's an issue in this file and the annex b section so I'll note it and come back to it.